index/secondary_index_manager.cc

```diff
@@ -144,7 +145,15 @@ view_ptr secondary_index_manager::create_view_for_index(const index_metadata& im
         builder.with_column(index_target->name(), index_target->type, column_kind::partition_key);
         // Additional token column is added to ensure token order on secondary index queries
         bytes token_column_name = get_available_token_column_name(*schema);
-        builder.with_computed_column(token_column_name, bytes_type, column_kind::clustering_key, std::make_unique<token_column_computation>());
+        column_computation_ptr computation;
+        if (service::get_local_storage_service().db().local().features().cluster_supports_correct_idx_token_in_secondary_index()) {
+            computation = std::make_unique<unsigned_token_column_computation>();
+        } else {
+            // FIXME(pgrabowski): this legacy code is here for backward compatibility and should be removed
+            // once "supports_correct_idx_token_in_secondary_index" is supported by every node
+            computation = std::make_unique<token_column_computation>();
```

Comment:
That's what I was talking about above. This `token_column_computation` name sounds correct, so users might get tempted to use it instead of `unsigned_token_column_computation`, which sounds weird. Let's rename the correct version to `token_column_computation` and invent an ugly, deterring name for the old version - e.g. `legacy::token_column_computation`.

Reply:
Renamed to `legacy_token_column_computation` in new commits.